docs/en/UI/Angular/Component-Replacement.md

```diff
@@ -77,6 +77,140 @@ export class AppComponent {
 
 ![Layout Components](./images/layout-components.png)
 
+### How to Add a New Layout Component
+
+To add a new layout component, you need to follow these steps:
+
+#### Step 1: Create a New Angular Component
+
+First, you need to create a new Angular component. This component should have a 'router-outlet' for dynamic content loading. You can create a new component using the Angular CLI. Run the following command in your terminal:
+
+```bash
+ng generate component new-layout
+```
+This command will create a new component named `new-layout`. Now, open the new-layout.component.html file and add a `router-outlet` to it:
+
+```html
+  <router-outlet></router-outlet>
+```
+This 'router-outlet' will act as a placeholder that Angular dynamically fills based on the current router state.
+
+note: (don't forget: you should add the app in the app.module.ts file)
+
+#### Step 2: Define a Variable for the Layout Component
+
+Although this step is optional, it can be useful if you're going to use the layout component's value multiple times. You can define a variable for the layout component like this:
+
+```javascript
+export const eCustomLayout = {
+    key: 'CustomLayout',
+    component: 'CustomLayoutComponent',
+};
+```
```

Comment:
Where will I define this variable ?

Reply:
I thoutht that but it is depend on the users's desire. There is no rule for that.